### PR TITLE
Improve documentation 1 to implement 1

### PR DESCRIPTION
**Pull request by schettn**
**Checklist:**

 
 - [x] Have you added an explanation of what your changes do and why you'd like them to be included?
 - [x] Have you updated or added documentation for the change?
 - [x] Have you tested your changes with successful results?

**Type of Changes:**

 
 - [ ] Bug fix (non-breaking change which fixes an issue)
 - [ ] New feature (non-breaking change which adds functionality)
 - [x] Documentation (non-breaking change which adds documentation)
 - [ ] Breaking change (fix or feature that would cause existing functionality to change)

**What is the current behavior? (link to any open issues here)**
- Currently the documentation does not follow our standard. #55
- Currently the anchor definitions do not follow our standard. #68

**What is the new behavior (if this is a feature change)?**
- Now the documentation does follow our standard. #55
- Now the anchor definitions do follow our standard. #68

**Other information**:
- Ref: https://githb.com/snek-at/tonic/
- Ref: 🐍

### CODE REVIEW

**Comment by pinterid** [comment]
The type of change should be set before merging.

File: src/reducer/database/helper/calendar.ts [hunk 1]
```diff
@@ -77,9 +95,11 @@ function generateCalendarStructure(startDate: string, endDate: string) {
 /**
  * Fill all days of a calendar object with colors.
  *
- * @param calendar Calendar object generated by "generateCalendarStructure".
- * @param busiestDayTotal The total amount of contributions for calculating the color spectrum.
- * @returns {object} A calendar object where all days have a specific color based on their contributions.
+ * @param calendar Calendar object generated by "generateCalendarStructure"
+ * @param busiestDayTotal The total amount of contributions for calculating the
+ *                        color spectrum
```
Comment (pinterid) [inline src/reducer/database/helper/calendar.ts:100]: Missing dot.

File: src/reducer/database/helper/calendar.ts [hunk 2]
```diff
@@ -77,9 +95,11 @@ function generateCalendarStructure(startDate: string, endDate: string) {
...
+ * @returns {object} A calendar object where all days have a specific color based
+ *                   on their contributions
```
Comment (pinterid) [inline src/reducer/database/helper/calendar.ts:102]: Missing dot.

File: src/reducer/database/helper/general.ts [hunk 3]
```diff
@@ -2,11 +2,13 @@
...
- * @todo Implement a way to remove objects/arrays from a object.
- * @param obj A object to squeeze.
- * @param keys A list of object keys to filter by.
- * @returns {object} The filtered object.
- * @description Removes every key of an object that does not appear in the key list.
+ * @function
+ * @param obj A object to squeeze
+ * @param keys A list of object keys to filter by
+ * @returns {object} The filtered object
+ * @description Removes every key of an object that does not appear in the
```
Comment (pinterid) [inline src/reducer/database/helper/general.ts:9]: But "@description" below "@todo".

File: src/reducer/database/helper/general.ts [hunk 4]
```diff
@@ -20,7 +22,9 @@ function squeezer(obj: any, keys?: string[], exclude: string[] = ["objects"]) {
   for (let index = 0; index < keysToRemove.length; index++) {
     /**
      * Proxy to neutralize Generic Object Injection Sink.
-     * @todo Validate the proxy workaround for the injection vulnerability via a penetration test.
+     *
+     * @todo Validate the proxy workaround for the injection vulnerability via
+     *       a penetration test
```
Comment (pinterid) [inline src/reducer/database/helper/general.ts:27]: Missing dot.

File: src/reducer/helper/talks.ts [hunk 5]
```diff
@@ -5,13 +5,14 @@ import { Repository, Talk } from "../database/models";
 //#endregion
 
 //#region > Interfaces
-/** @interface Talk defines the structure of a talk object */
+/** @interface Talk defines the structure of a talk object. */
```
Comment (pinterid) [inline src/reducer/helper/talks.ts:8]: Remove dot.

File: src/utils/oauth/GithubProvider.ts [hunk 6]
```diff
@@ -1,5 +1,6 @@
 //#region > Imports
-//> Interfaces
+//#PACKAGE "react-very-simple-oauth"
+//## npm i "react-very-simple-oauth"@1.1.1
```
Comment (pinterid) [inline src/utils/oauth/GithubProvider.ts:3]: Change "i" to "install".

File: src/utils/oauth/GithubProvider.ts [hunk 7]
```diff
@@ -39,9 +41,9 @@ const GithubProvider: IProvider<boolean> = {
   /**
    * Extracts Error
```
Comment (pinterid) [inline src/utils/oauth/GithubProvider.ts:42]: Missing dot.

File: src/reducer/database/helper/calendar.ts [hunk 8]
```diff
@@ -1,46 +1,64 @@
...
  *
- * @param startDate Date to begin the calendar with.
- * @param endDate Date to end the calendar with.
- * @returns {object} A calendar object with a structure containing weeks with empty days.
- * @description Generates a calendar object containing empty days from startDate to endDate.
+ * @function
+ * @param startDate Date to begin the calendar with
+ * @param endDate Date to end the calendar with
+ * @returns {object} A calendar object with a structure containing weeks with
+ *                   empty days
```
Comment (pinterid) [inline src/reducer/database/helper/calendar.ts:59]: Missing dot.

File: src/reducer/database/helper/calendar.ts [hunk 9]
```diff
@@ -1,46 +1,64 @@
...
+ * @description Generates a calendar object containing empty days from
+ *              startDate to endDate
```
Comment (pinterid) [inline src/reducer/database/helper/calendar.ts:61]: Missing dot.

File: src/reducer/database/models.ts [hunk 10]
```diff
@@ -385,11 +382,10 @@ class Repository extends osm.models.RepositorySO implements IRepository {
   }
 
   /**
-   * Create a language entry.
+   * Create a language entry within this repository.
    *
-   * @param fields Language data.
-   * @returns {Language} A language entry object.
-   * @description Create a language entry within this repository.
```
Comment (kleberbaum) [inline src/reducer/database/models.ts:392]: Please add a rule to snektonic.

File: src/reducer/database/models.ts [hunk 11]
```diff
@@ -431,15 +425,15 @@ class Repository extends osm.models.RepositorySO implements IRepository {
   }
 
   /**
-   * @todo Implement a save functionality.
-   * @description Saves this osm model.
+   * @todo Implement a save functionality
+   * @description Saves this osm model
```
Comment (kleberbaum) [inline src/reducer/database/models.ts:429]: Please add a rule to snektonic.

File: src/reducer/database/models.ts [hunk 12]
```diff
@@ -822,7 +812,10 @@ class Statistic extends osm.models.StatisticSO implements IStatistic {
       }
     }
 
-    /* Create empty calendar entry for busiest day if there is no real busiest day */
+    /*
+     * Create empty calendar entry for busiest day, if there is no real
+     * busiest day-
```
Comment (kleberbaum) [inline src/reducer/database/models.ts:817]: lmao-

File: src/utils/github/queries/index.ts [hunk 13]
```diff
@@ -1,21 +1,25 @@
...
+// Contains all queries for github graphql api
 import * as queries from "./data";
 //#endregion
 
 //#region > Interfaces
 /**
- * @interface Profile defines the structure of the profile object which is requiered to generate the calendar query.
- * @description Only createdAt is defined due to no more information is needed to generate the calendar query.
+ * @interface Profile defines the structure of the profile object which is required
+ *                   to generate the calendar query.
```
Comment (kleberbaum) [inline src/utils/github/queries/index.ts:15]: Please fix indentation.

File: src/reducer/database/osm/models.ts [hunk 14]
```diff
@@ -140,67 +140,67 @@ abstract class PlatformSO extends BaseSO {
...
   //> Methods
   /**
    * @static
-   * @returns {string} A table name.
-   * @description Returns the database table name of this osm model.
+   * @returns {string} A table name
+   * @description Returns the database table name of this osm model
    */
-  tableName() {
+  tableName(): string {
```
Comment (kleberbaum) [inline src/reducer/database/osm/models.ts:203]: Shouldn't be a change in a a doku branch.

File: src/reducer/database/osm/models.ts [hunk 15]
```diff
@@ -278,10 +278,10 @@ abstract class MemberSO extends BaseSO {
...
```
Comment (kleberbaum) [inline src/reducer/database/osm/models.ts:284]: Shouldn't be a change in a a doku branch.

File: src/reducer/database/osm/models.ts [hunk 16]
```diff
@@ -359,47 +359,47 @@ abstract class RepositorySO extends BaseSO {
...
```
Comment (kleberbaum) [inline src/reducer/database/osm/models.ts:402]: Shouldn't be a change in a a doku branch.

File: src/reducer/database/osm/models.ts [hunk 17]
```diff
@@ -477,10 +477,10 @@ abstract class RepositoryHasMemberSO extends BaseSO {
...
```
Comment (kleberbaum) [inline src/reducer/database/osm/models.ts:483]: Shouldn't be a change in a a doku branch.

File: src/reducer/database/osm/models.ts [hunk 18]
```diff
@@ -559,10 +559,10 @@ abstract class LanguageSO extends BaseSO {
...
```
Comment (kleberbaum) [inline src/reducer/database/osm/models.ts:565]: Shouldn't be a change in a a doku branch.

File: src/reducer/database/osm/models.ts [hunk 19]
```diff
@@ -641,10 +641,10 @@ abstract class PlatformHasRepositorySO extends BaseSO {
...
```
Comment (kleberbaum) [inline src/reducer/database/osm/models.ts:647]: Shouldn't be a change in a a doku branch.

File: src/reducer/database/osm/models.ts [hunk 20]
```diff
@@ -726,33 +726,33 @@ abstract class OrganizationSO extends BaseSO {
...
```
Comment (kleberbaum) [inline src/reducer/database/osm/models.ts:755]: Shouldn't be a change in a a doku branch.

File: src/reducer/database/osm/models.ts [hunk 21]
```diff
@@ -830,10 +830,10 @@ abstract class OrganizationHasMemberSO extends BaseSO {
...
```
Comment (kleberbaum) [inline src/reducer/database/osm/models.ts:836]: Shouldn't be a change in a a doku branch.

File: src/reducer/database/osm/models.ts [hunk 22]
```diff
@@ -915,10 +915,10 @@ abstract class PlatformHasOrganizationSO extends BaseSO {
...
```
Comment (kleberbaum) [inline src/reducer/database/osm/models.ts:921]: Shouldn't be a change in a a doku branch.

File: src/reducer/database/osm/models.ts [hunk 23]
```diff
@@ -1114,10 +1114,10 @@ abstract class StatisticSO extends BaseSO {
...
```
Comment (kleberbaum) [inline src/reducer/database/osm/models.ts:1120]: Shouldn't be a change in a a doku branch.

File: src/reducer/database/osm/models.ts [hunk 24]
```diff
@@ -1194,10 +1194,10 @@ abstract class StreakSO extends BaseSO {
...
```
Comment (kleberbaum) [inline src/reducer/database/osm/models.ts:1200]: Shouldn't be a change in a a doku branch.

File: src/reducer/database/osm/models.ts [hunk 25]
```diff
@@ -1364,19 +1363,19 @@ abstract class CalendarSO extends BaseSO {
...
```
Comment (kleberbaum) [inline src/reducer/database/osm/models.ts:1378]: Shouldn't be a change in a a doku branch.

File: src/reducer/database/osm/models.ts [hunk 26]
```diff
@@ -1453,10 +1452,10 @@ abstract class ContributionSO extends BaseSO {
...
```
Comment (kleberbaum) [inline src/reducer/database/osm/models.ts:1458]: Shouldn't be a change in a a doku branch.

File: src/toolbox/GuidGenerator.ts [hunk 27]
```diff
@@ -1,10 +1,10 @@
...
 /**
- * @function Generate random guid string.
- * @returns {string} A guid string.
- * @description This function generates a random guid string.
+ * @function Generate random guid string
+ * @returns {string} A guid string
+ * @description This function generates a random guid string
  */
-function GuidGenerator() {
+function GuidGenerator(): string {
```
Comment (kleberbaum) [inline src/toolbox/GuidGenerator.ts:7]: Shouldn't be a change in a a doku branch.

File: src/utils/gitlab/paths.ts [hunk 28]
```diff
@@ -1,12 +1,15 @@
 //#region > Paths
-/* Define Gitlab website paths */
+/* Define GitLab website paths */
 const root = "/";
-const home = (username: string) => root  + username;
-const currentCalendar = (username: string) => root + "users/" + username + "/calendar";
+const home = (username: string) => root + username;
```
Comment (kleberbaum) [inline src/utils/gitlab/paths.ts:4]: Shouldn't be a change in a a doku branch.

File: src/utils/gitlab/paths.ts [hunk 29]
```diff
@@ -1,12 +1,15 @@
...
+const currentCalendar = (username: string) =>
```
Comment (kleberbaum) [inline src/utils/gitlab/paths.ts:5]: Shouldn't be a change in a a doku branch.

File: src/utils/gitlab/paths.ts [hunk 30]
```diff
@@ -1,12 +1,15 @@
...
+  root + "users/" + username + "/calendar";
 const activity = (username: string) => root + username + "?limit=2147483647";
 const atom = (username: string) => root + username + ".atom?limit=2147483647";
-const groups = (username: string) => root + "users/" + username + "/groups.json";
-const projects = (username: string) => root + "users/" + username + "/projects.json";
+const groups = (username: string) =>
```
Comment (kleberbaum) [inline src/utils/gitlab/paths.ts:9]: Shouldn't be a change in a a doku branch.

File: src/utils/gitlab/paths.ts [hunk 31]
```diff
@@ -1,12 +1,15 @@
...
+  root + "users/" + username + "/groups.json";
+const projects = (username: string) =>
```
Comment (kleberbaum) [inline src/utils/gitlab/paths.ts:11]: Shouldn't be a change in a a doku branch.

File: src/utils/talks/converter.ts [hunk 32]
```diff
@@ -13,7 +13,7 @@ import { Talk } from "../../reducer/database/models";
 async function run(rawData: any) {
   rawData.talks.forEach((item: any) => {
     let displayUrl = "https://docs.google.com/viewer?embedded=true&url=";
-    let downloadUrl;
+    let downloadUrl: string;
```
Comment (kleberbaum) [inline src/utils/talks/converter.ts:16]: Shouldn't be a change in a a doku branch.

File: src/reducer/database/models.ts (hunk 10 above)
Comment (schettn) [reply src/reducer/database/models.ts:392]: Yep, but it is ok for now.

File: src/utils/gitlab/paths.ts [hunk 33]
```diff
@@ -1,7 +1,7 @@
 //#region > Paths
-/* Define Gitlab website paths */
+/* Define GitLab website paths */
 const root = "/";
-const home = (username: string) => root  + username;
```
Comment (kleberbaum) [inline src/utils/gitlab/paths.ts:4]: `const home = (username: string) => root  + username;`

File: src/utils/gitlab/paths.ts [hunk 34]
```diff
@@ -1,7 +1,7 @@
...
+const home = (username: string) => root + username;
```
Comment (schettn) [inline src/utils/gitlab/paths.ts:4]:
```suggestion
const home = (username: string) => root  + username;
```

**Pull request merged**